file_transfer_agent_test.go

```diff
@@ -597,3 +597,126 @@ func TestUploadWhenFilesystemReadOnlyError(t *testing.T) {
 		t.Fatalf("should error when creating the temporary directory. Instead errored with: %v", err)
 	}
 }
+
+func TestCustomTmpDirPath(t *testing.T) {
+	tmpDir, err := os.MkdirTemp("/tmp", "")
+	if err != nil {
+		t.Fatalf("cannot create temp directory: %v", err)
+	}
+	defer os.RemoveAll(tmpDir)
+	uploadFile := filepath.Join(tmpDir, "data.txt")
+	f, err := os.Create(uploadFile)
+	if err != nil {
+		t.Error(err)
+	}
+	f.WriteString("test1,test2\ntest3,test4\n")
+	f.Close()
+
+	uploadMeta := &fileMetadata{
+		name:              "data.txt.gz",
+		stageLocationType: "local",
+		noSleepingTime:    true,
+		client:            local,
+		sha256Digest:      "123456789abcdef",
+		stageInfo: &execResponseStageInfo{
+			Location:     tmpDir,
+			LocationType: "local",
+		},
+		dstFileName: "data.txt.gz",
+		srcFileName: uploadFile,
+		overwrite:   true,
+		options: &SnowflakeFileTransferOptions{
+			MultiPartThreshold: dataSizeThreshold,
+		},
+	}
+
+	downloadFile := filepath.Join(tmpDir, "download.txt")
+	downloadMeta := &fileMetadata{
+		name:              "data.txt.gz",
+		stageLocationType: "local",
+		noSleepingTime:    true,
+		client:            local,
+		sha256Digest:      "123456789abcdef",
+		stageInfo: &execResponseStageInfo{
+			Location:     tmpDir,
+			LocationType: "local",
+		},
+		srcFileName: "data.txt.gz",
+		dstFileName: downloadFile,
+		overwrite:   true,
+		options: &SnowflakeFileTransferOptions{
+			MultiPartThreshold: dataSizeThreshold,
+		},
+	}
+
+	sfa := snowflakeFileTransferAgent{
+		sc: &snowflakeConn{
+			cfg: &Config{
+				TmpDirPath: tmpDir,
+			},
+		},
+		stageLocationType: local,
+	}
+	_, err = sfa.uploadOneFile(uploadMeta)
+	if err != nil {
+		t.Fatal(err)
+	}
+	_, err = sfa.downloadOneFile(downloadMeta)
```

Comment:
shouldn't we be interested in results? not only in errors?

Reply:
No, it doesn't matter. The purpose of this test is to test if custom path does not cause any error.